include/exadg/time_integration/time_int_abm_base.h

```diff
@@ -0,0 +1,266 @@
+#include <exadg/time_integration/push_back_vectors.h>
+#include <exadg/time_integration/time_int_multistep_base.h>
+
+namespace ExaDG
+{
+template<typename Operator, typename VectorType>
+class TimeIntABMBase : public TimeIntMultistepBase
+{
+  using Number = typename VectorType::value_type;
+
+public:
+  TimeIntABMBase(std::shared_ptr<Operator> pde_operator_in,
+                 double const              start_time_,
+                 double const              end_time_,
+                 unsigned int const        max_number_of_time_steps_,
+                 unsigned int const        order_,
+                 bool const                start_with_low_order_,
+                 bool const                adaptive_time_stepping_,
+                 RestartData const &       restart_data_,
+                 MPI_Comm const &          mpi_comm_,
+                 bool const                is_test_)
+    : TimeIntMultistepBase(start_time_,
+                           end_time_,
+                           max_number_of_time_steps_,
+                           order_,
+                           start_with_low_order_,
+                           adaptive_time_stepping_,
+                           restart_data_,
+                           mpi_comm_,
+                           is_test_),
+      pde_operator(pde_operator_in),
+      // order of predictor can be chosen one below order of corrector
+      ab(order_ - 1, start_with_low_order_),
+      am(order_, start_with_low_order_),
+      vec_evaluated_operators(order_ - 1)
+  {
+    AssertThrow(order_ >= 1,
+                dealii::ExcMessage("Oder of ABM time integrator has to be at least 1."));
+  }
+
+  void
+  print_iterations() const
+  {
+    // explicit time integration -> one iteration
+    print_list_of_iterations(pcout, {"Adams-Bashforth-Moulton"}, {1});
+  }
+
+  virtual bool
+  print_solver_info() const = 0;
+
+  void
+  ale_update()
+  {
+    AssertThrow(false, dealii::ExcMessage("not yet implemented"));
+  }
+
+  VectorType const &
+  get_solution() const
+  {
+    return solution;
+  }
+
+private:
+  void
+  update_time_integrator_constants() final
+  {
+    ab.update(time_step_number, adaptive_time_stepping, time_steps);
+    am.update(time_step_number, adaptive_time_stepping, time_steps);
+  }
+
+  void
+  allocate_vectors() final
+  {
+    pde_operator->initialize_dof_vector(solution);
+    pde_operator->initialize_dof_vector(prediction);
+
+    pde_operator->initialize_dof_vector(evaluated_operators_np);
+    for(auto & evaluated_operator : vec_evaluated_operators)
+      pde_operator->initialize_dof_vector(evaluated_operator);
+  }
+
+  void
+  initialize_current_solution() final
+  {
+    pde_operator->prescribe_initial_conditions(solution, get_time());
+  }
+
+  void
+  initialize_former_multistep_dof_vectors() final
+  {
+    if(start_with_low_order)
+    {
+      if(vec_evaluated_operators.size() > 0)
+        evaluate_operator(vec_evaluated_operators[0], solution, get_time());
```

Comment:
What is the content of `solution` at this point? Only one line below the initial condition is prescribed. I am irritated.

Reply:
`solution` is correctly initialized at this point. The initial condition does not have to be prescribed again. [d295d6a](https://github.com/exadg/exadg/pull/585/commits/d295d6a60072842c980ccebd0eb64ee62e41811b)